src/commands/Minion/slayertask.ts

```diff
@@ -0,0 +1,195 @@
+			msg.author.skillLevel(SkillsEnum.Magic),
+			msg.author.skillLevel(SkillsEnum.Range)
+		);
+		*/
+		const userCombatLevel = 126;
+		if (
+			(master.combatLvl && master.combatLvl > userCombatLevel) ||
+			(master.slayerLvl && master.slayerLvl > msg.author.skillLevel(SkillsEnum.Slayer)) ||
+			(master.questPoints && master.questPoints > msg.author.settings.get(UserSettings.QP))
+		) {
+			throw `You need a combat level of ${master.combatLvl}, a slayer level of ${
+				master.slayerLvl
+			} and ${master.questPoints} quest points to use this master! 
+You're only ${userCombatLevel} combat, ${msg.author.skillLevel(
+				SkillsEnum.Slayer
+			)} slayer and ${msg.author.settings.get(UserSettings.QP)} questpoints.`;
+		}
+		let filteredTaskList;
+		// Filter by slayer level
+		filteredTaskList = listOfTasks.filter(
+			task => task.slayerLvl && task.slayerLvl <= msg.author.skillLevel(SkillsEnum.Slayer)
+		);
+		// Filter by combat level
+		filteredTaskList = filteredTaskList.filter(
+			task => task.combatLvl && task.combatLvl <= userCombatLevel
+		);
+		// Filter by questpoints, not implemented
+		/*
+		filteredTaskList = filteredTaskList.filter(
+			task =>
+				task.QP && task.QP <= msg.author.settings.get(UserSettings.QP)
+				
+		);
+		*/
+		// Filter by default unlock
+		filteredTaskList = filteredTaskList.filter(task => task.unlocked === true);
+		// Filter by block list (Make Task based)
+		/*
+		filteredTaskList = filteredTaskList.filter(task => !msg.author.blockList.includes(task));
+		*/
+		// Filter by unlocks (Make Task based)
+		/*
+		const filteredByUnlocked = listOfTasks.filter(task => msg.author.unlockedList.includes(task));
+		filteredTasks = filteredTasks.concat(filteredByUnlocked);
+		*/
+
+		let totalweight = 0;
+		for (let i = 0; i < filteredTaskList.length; i++) {
+			totalweight += filteredTaskList[i].weight;
+		}
+		if (filteredTaskList.length === 0) {
+			throw `You don't have a high enough Slayer level to get a task from that Master.`;
+		}
+		let number = rand(1, totalweight);
+		for (let i = 0; i < filteredTaskList.length; i++) {
+			number -= filteredTaskList[i].weight;
+			if (number <= 0) {
+				const slayerMonster = filteredTaskList[i];
+				/*
+				if (slayerMonster.name === 'Boss') {
+					const filteredBossTasks = bossTasks.filter(
+						task =>
+							task.slayerLvl &&
+							task.slayerLvl <= msg.author.skillLevel(SkillsEnum.Slayer)
+					);
+					slayerMonster = filteredBossTasks[rand(0, filteredBossTasks.length)];
+				}
+				*/
+				const minQuantity = slayerMonster.amount[0];
+				const maxQuantity = slayerMonster.amount[1];
+				const quantity = Math.floor(rand(minQuantity, maxQuantity));
+				const newSlayerInfo = msg.author.slayerInfo;
+				newSlayerInfo.hasTask = true;
+				newSlayerInfo.currentTask = slayerMonster;
+				newSlayerInfo.quantityTask = quantity;
+				newSlayerInfo.remainingQuantity = quantity;
+				newSlayerInfo.currentMaster = master.masterId;
```

Comment:
```suggestion
				const newSlayerInfo = {
					...msg.author.slayerInfo,
					hasTask: true,
					currentTask: slayerMonster,
					quantityTask: quantity,
					remainingQuantity: quantity,
					currentMaster: master.masterId
				};
```

Would also recommend changing `quantityTask` to `taskQuantity`